leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/impl/NamedThreadFactory.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Alexander Ellwein, Daniel Maier (Bosch Software Innovations GmbH)
+ *                                - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue.impl;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Implementation of {@link ThreadFactory} that sets thread names according to given name format. All threads are
+ * created by {@link Executors#defaultThreadFactory() #newThread(Runnable)}.
+ */
+final class NamedThreadFactory implements ThreadFactory {
```

Comment:
I was also thinking in the same line. See my comments above. Shall I move it to org.eclipse.leshan.util as part of this PR itself?